### DISCUSSION TITLE
参考文献集中定制需求

### DISCUSSION THREAD

**Discussion by sikouhjw** [sikouhjw/gdutthesis#68] [Announcements]
首先现在的方案是完全符合国标的，一切修改的行为都是在违反国标。

如果导师要求修改，可以在下面寻找解决方案，或者在下面提出需求。

**Comment by sikouhjw** [comment]
# 需求：将作者的姓名从『全大写』改成『大小写由输入信息确定不做改变』。

## bibtex

参考 `texdoc gbt7714`。

### 本地

按以下步骤修改：
1. 命令行输入 `kpsewhich gbt7714-numerical.bst`；
2. 将返回路径代表的文件复制到工作目录下；
3. 用文本编辑器打开复制后的 `gbt7714-numerical.bst`；
4. 找到 `uppercase.name` 项，将其 `#1` 改为 `#0`，删除辅助文件再重新编译。

### Overleaf

按以下步骤修改：
1. 到 github[^1]或者 ctan[^2]下载 `gbt7714-numerical.bst`；
2. 用文本编辑器打开下载的 `gbt7714-numerical.bst`；
3. 找到 `uppercase.name` 项，将其 `#1` 改为 `#0`；
4. 将修改后的 `gbt7714-numerical.bst` 上传到 Overleaf 上，删除辅助文件再重新编译。

[^1]: https://github.com/zepinglee/gbt7714-bibtex-style/releases/latest
[^2]: https://www.ctan.org/pkg/gbt7714

## biblatex

参考 `texdoc biblatex-gb7714-2015`。

按以下步骤修改：
1. 删除全部辅助文件；
2. `bib-backend` 设置为 `bib-backend = {biblatex}`；
3. 在导言区添加 `\PassOptionsToPackage{gbnamefmt=lowercase}{biblatex}`；
4. 用 `latexmk -xelatex` 或者 `xelatex biber xelatex xelatex` 编译。

**Comment by xkwxdyy** [comment]
> 首先现在的方案是完全符合国标的，一切修改的行为都是在违反国标。
> 
> 如果导师要求修改，可以在下面寻找解决方案，或者在下面提出需求。

好想法，但是有多少学校真是完全按国标来呢（

**Reply by sikouhjw** [reply]
好问题，规范写的确实不是 2015 版的国标，但本科的从 87 改成了 2015，研究生的还没改，但是检测系统也要求用 2015 的参考文献，所以模板用 2015 应该没人能反对（

**Comment by sikouhjw** [comment]
# 需求：既想保留 `doi`、`url`，又要去掉『载体标识代码』。

## bibtex

参考 `texdoc gbt7714`。

### 本地

按以下步骤修改：
1. 命令行输入 `kpsewhich gbt7714-numerical.bst`；
2. 将返回路径代表的文件复制到工作目录下；
3. 用文本编辑器打开复制后的 `gbt7714-numerical.bst`；
4. 找到 `show.medium.type` 项，将其 `#1` 改为 `#0`，删除辅助文件再重新编译。

### Overleaf

按以下步骤修改：
1. 到 github[^1]或者 ctan[^2]下载 `gbt7714-numerical.bst`；
2. 用文本编辑器打开下载的 `gbt7714-numerical.bst`；
3. 找到 `show.medium.type` 项，将其 `#1` 改为 `#0`；
4. 将修改后的 `gbt7714-numerical.bst` 上传到 Overleaf 上，删除辅助文件再重新编译。

[^1]: https://github.com/zepinglee/gbt7714-bibtex-style/releases/latest
[^2]: https://www.ctan.org/pkg/gbt7714

## biblatex

参考 `texdoc biblatex-gb7714-2015`。

按以下步骤修改：
1. 删除全部辅助文件；
2. `bib-backend` 设置为 `bib-backend = {biblatex}`；
3. 在导言区添加 `\PassOptionsToPackage{gbmedium=false}{biblatex}`；
4. 用 `latexmk -xelatex` 或者 `xelatex biber xelatex xelatex` 编译。

**Comment by sikouhjw** [comment]
# 需求：在『名字』缩写后加点 `.`。

## bibtex

参考《如何在姓氏的简写名后加点 `.`？》[^4]。

[^4]: https://github.com/zepinglee/gbt7714-bibtex-style/issues/125

### 本地

按以下步骤修改：
1. 命令行输入 `kpsewhich gbt7714-numerical.bst`；
2. 将返回路径代表的文件复制到工作目录下；
3. 用文本编辑器打开复制后的 `gbt7714-numerical.bst`；
4. 找到第 593 行的 `{ t #1 "{vv~}{ll}{ f{~}}" format.name$`，将其改为 `{ t #1 "{vv~}{ll}{ f.}" format.name$`，删除辅助文件再重新编译。

### Overleaf

按以下步骤修改：
1. 到 github[^1]或者 ctan[^2]下载 `gbt7714-numerical.bst`；
2. 用文本编辑器打开下载的 `gbt7714-numerical.bst`；
3. 找到第 593 行的 `{ t #1 "{vv~}{ll}{ f{~}}" format.name$`，将其改为 `{ t #1 "{vv~}{ll}{ f.}" format.name$`，删除辅助文件再重新编译；
4. 将修改后的 `gbt7714-numerical.bst` 上传到 Overleaf 上，删除辅助文件再重新编译。

[^1]: https://github.com/zepinglee/gbt7714-bibtex-style/releases/latest
[^2]: https://www.ctan.org/pkg/gbt7714

## biblatex

参考《Latex参考文献作者名字简写如何加点号？》[^3]。

[^3]: https://www.zhihu.com/question/576026028/answer/2825597226

按以下步骤修改：
1. 删除全部辅助文件；
2. `bib-backend` 设置为 `bib-backend = {biblatex}`；
3. 在导言区添加
    ```tex
    \ExplSyntaxOn
    \__gdut_gadd_ltxhook:nn { env/document/begin }
      {
        \def\gbcaselocalset
          {
            \renewrobustcmd*  { \bibinitperiod  } { \adddot }
            \renewcommand*    { \revsdnamepunct } { }
          }
      }
    \ExplSyntaxOff
    ```
5. 用 `latexmk -xelatex` 或者 `xelatex biber xelatex xelatex` 编译。

**Comment by sikouhjw** [comment]
# 需求：修改 arXiv 参考文献样式。

gbt7714 宏包作者认为 arXiv 属于“档案”，即使用标识 `[A]`[^1]。

在 bib 写法方面，可以使用 arXiv 提供的写法，也可以使用 biblatex 的写法[^2]，在实践层面，谷歌学术导出的文献也可以正确识别。

如果认为 arXiv 不应当使用 `[A]`，那么可以根据以下部分修改。

## bibtex
https://github.com/zepinglee/gbt7714-bibtex-style/issues/155#issuecomment-1966353586

[^1]: https://github.com/zepinglee/gbt7714-bibtex-style/issues/89#issuecomment-950295152
[^2]: https://github.com/zepinglee/gbt7714-bibtex-style/issues/89#issuecomment-1265112843
